bin/convert_owl.py

```diff
@@ -0,0 +1,130 @@
+"""Convert OWL to FHIR"""
+import os
+import subprocess
+from argparse import ArgumentParser
+from typing import Dict
+
+import curies
+import requests
+from linkml_runtime.loaders import json_loader
+from oaklib.converters.obo_graph_to_fhir_converter import OboGraphToFHIRConverter
+from oaklib.datamodels.obograph import GraphDocument
+from oaklib.interfaces.basic_ontology_interface import get_default_prefix_map
+from urllib.parse import urlparse
+
+
+# Vars
+BIN_DIR = os.path.dirname(os.path.realpath(__file__))
+PROJECT_DIR = os.path.join(BIN_DIR, '..')
+CACHE_DIR = os.path.join(PROJECT_DIR, 'cache')
+OUTDIR = os.path.join(PROJECT_DIR, 'output')
+ROBOT_PATH = os.path.join(BIN_DIR, 'robot')
+
+
+# Functions
+def download(url: str, path: str, download_if_cached=True):
+    """Download file at url to local path
+
+    :param download_if_cached: If True and file at `path` already exists, download anyway."""
+    _dir = os.path.dirname(path)
+    if not os.path.exists(_dir):
+        os.makedirs(_dir)
+    if download_if_cached or not os.path.exists(path):
+        with open(path, 'wb') as f:
+            response = requests.get(url, verify=False)
+            f.write(response.content)
+
+
+def owl_to_obograph(inpath: str, outpath: str):
+    """Convert OWL to Obograph"""
+    # todo: Switch back to `bioontologies` when complete: https://github.com/biopragmatics/bioontologies/issues/9
+    # from bioontologies import robot
+    # parse_results: robot.ParseResults = robot.convert_to_obograph_local(inpath)
+    # graph = parse_results.graph_document.graphs[0]
+    args = ['java', '-jar', ROBOT_PATH + '.jar', 'convert', '-i', inpath, '-o', outpath, '--format', 'json']
+    result = subprocess.run(args, capture_output=True, text=True)
+    stderr, stdout = result.stderr, result.stdout
+    if stderr:
+        raise RuntimeError(stderr)
+    elif stdout and 'error' in stdout or 'ERROR' in stdout:
+        raise RuntimeError(stdout)
+
+
+# todo: When https://github.com/INCATools/ontology-access-kit/pull/374 is merged, can add include_all_predicates
+def obograph_to_fhir(inpath: str, out_dir: str, out_filename: str = None, include_all_predicates=False):
+    """Convert Obograph to FHIR"""
+    converter = OboGraphToFHIRConverter()
+    converter.curie_converter = curies.Converter.from_prefix_map(get_default_prefix_map())
+    gd: GraphDocument = json_loader.load(inpath, target_class=GraphDocument)
+    out_path = os.path.join(out_dir, out_filename)
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+    # converter.dump(gd, out_path, include_all_predicates=True)
+    converter.dump(gd, out_path)
+
+
+def owl_to_fhir(
```

Comment:
@ShahimEssaid This is the function you would want to import and tie to your CLI.